boa/src/exec/iteration/mod.rs

```diff
@@ -34,8 +34,22 @@ impl Executable for ForLoop {
             let result = self.body().run(interpreter)?;
 
             match interpreter.executor().get_current_state() {
-                InterpreterState::Break(_label) => {
-                    // TODO break to label.
+                InterpreterState::Break(label) => {
+                    // If a label is set we want to break the current block and still keep state as Break if the label is a block above
+                    if let Some(stmt_label) = &self.label {
+                        if let Some(brk_label) = label {
+                            // We have a label, but not for the current statement
+                            // break without resetting to executings
+                            if stmt_label.to_string() != *brk_label {
```

Comment:
Why do we do `.to_string()` here? It seems like an unnecessary string allocation.

Reply:
Yeah this was annoying, `stmt_label` is a `Box<str>` and `brk_label` is a `&String`. 
Ideally `brk_label` should be a `Box<str>` but it was not possible to make that happen, i can't remember why i will have to look at it again to find out

Reply:
I don't we why we can't change `InterpreterState::Break(Option<String>)` to be `InterpreterState::Break(Option<Box<str>>)` this should also be more space efficient.

Reply:
As far as I understand, `Box<str>` and `&String` can be compared by doing:
```rust
if &stm_label != brk_label.as_str() {
    todo!()
}
```

And the `&` in the `stm_label` might be optional, or it might require a `stm_label.as_ref()`.

Reply:
> I don't we why we can't change InterpreterState::Break(Option<String>) to be InterpreterState::Break(Option<Box<str>>) this should also be more space efficient.

Yep thats what i originally started with

Reply:
The problem starts here: https://github.com/boa-dev/boa/blob/master/boa/src/exec/break_node/mod.rs#L14
I can't change this to `Box<str>` because ownership of the string will be moved (and Break needs to keep ownership of the label). So at the very minimum it needs to be `&Box<str>` but this isn't easy because it means the whole of `Interpreter` and `Context` will need lifetimes added to it because it's holding a burrowed value.

Reply:
> As far as I understand, `Box<str>` and `&String` can be compared by doing:
> 
> ```rust
> if &stm_label != brk_label.as_str() {
>     todo!()
> }
> ```
> 
> And the `&` in the `stm_label` might be optional, or it might require a `stm_label.as_ref()`.

That doesn't work
![image](https://user-images.githubusercontent.com/936006/94297865-1d7d1700-ff5d-11ea-8963-056d745d8221.png)

Reply:
> The problem starts here: [https://github.com/boa-dev/boa/blob/master/boa/src/exec/break_node/mod.rs#L14](https://github.com/boa-dev/boa/blob/master/boa/src/exec/break_node/mod.rs?rgh-link-date=2020-09-25T17%3A27%3A55Z#L14)
> I can't change this to `Box<str>` because ownership of the string will be moved (and Break needs to keep ownership of the label). So at the very minimum it needs to be `&Box<str>` but this isn't easy because it means the whole of `Interpreter` and `Context` will need lifetimes added to it because it's holding a burrowed value.
> 
> The reason it works fine now is because we're creating a new string everytime we run Break. Another option would be to wrap it in a Box instead

Just change it to `Box::from`:
```
            .set_current_state(InterpreterState::Break(self.label().map(Box::from)));
```
Box<str> implements From for &str